src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderPublisher.java

```diff
@@ -346,6 +367,8 @@ public void onNext(BuildResponseItem item) {
             final String imageId;
             try(final DockerClient client = getClientWithNoTimeout()) {
                 imageId = client.buildImageCmd(tar)
+                        .withNoCache(noCache)
```

Comment:
FYI _usually_ I'd want to guarantee that "not ticking the new checkbox" would mean that no new code got called at all (i.e. the `withNoCache` method & `withPull` method wouldn't get called at all); that way, there would be no chance that existing users who didn't ask for new stuff could be affected by any bugs in the new stuff.
However, I've (just now) checked the `dockerjava` code and it looks like its `AbstrDockerCmdExec.bool` method treats both `null` and `false` as equivalent so, in this particular case, we can get away with _always_ calling `withNoCache` and `withPull`.